test/hotspot/jtreg/vmTestbase/nsk/jdb/stop_at/stop_at002/stop_at002.java

```diff
@@ -101,25 +94,62 @@ protected void runCases() {
             success = false;
         }
 
-        jdb.contToExit(3);
+        if (!checkBreakpointHit(DEBUGGEE_LOCATION1)) {
+            success = false;
+        }
+
+        if (!checkBreakpointHit(DEBUGGEE_LOCATION2)) {
+            success = false;
+        }
+
+        jdb.contToExit(1);
     }
 
-    private boolean checkStop (String location) {
+    private boolean checkStop(String location) {
         Paragrep grep;
         String[] reply;
         String found;
-        boolean result = true;
 
         log.display("Trying to set breakpoint at line: " + location);
         reply = jdb.receiveReplyFor(JdbCommand.stop_at + location);
 
         grep = new Paragrep(reply);
-        found = grep.findFirst(FAILURE_PATTERN);
+        found = grep.findFirst("Deferring breakpoint " + location);
+        if (found.length() == 0) {
+            log.complain("jdb failed to setup deferred breakpoint at line: " + location);
+            return false;
+        }
+
+        return true;
+    }
+
+    private boolean checkBreakpointHit(String location) {
+        Paragrep grep;
+        String[] reply;
+        String found;
+
+        log.display("continuing to breakpoint at line: " + location);
+        reply = jdb.receiveReplyFor(JdbCommand.cont);
+        grep = new Paragrep(reply);
+
+        found = grep.findFirst("Unable to set deferred breakpoint");
         if (found.length() > 0) {
-            log.complain("jdb failed to set line breakpoint at line: " + found);
-            result = false;
+            log.complain("jdb failed to set deferred breakpoint at line: " + location);
+            return false;
+        }
+
+        found = grep.findFirst("Set deferred breakpoint " + location);
+        if (found.length() == 0) {
+            log.complain("jdb failed to set deferred breakpoint at line: " + location);
+            return false;
+        }
+
+        found = grep.findFirst("Breakpoint hit: \"thread=main\", ");
```

Comment:
Q: Is it possible to check for the exact breakpoint location here?

Reply:
The problem is in the test we have:

`    static final String DEBUGGEE_LOCATION2 = DEBUGGEE_CLASS + "$Inner$MoreInner:78";`

And the jdb output is:

`reply[2]: Breakpoint hit: "thread=main", nsk.jdb.stop_at.stop_at002.stop_at002a$Inner$MoreInner.foo(), line=78 bci=0`

So I can't just search for DEBUGGEE_LOCATION2 in the output. I could do further factoring of DEBUGGEE_LOCATION2 so it is easier to construct a search pattern that would work, but I don't feel it is worth it.

Reply:
Okay, thanks.